vision/imageloader/imageloader.go

```diff
@@ -50,6 +50,11 @@ type ImageLoader struct {
 	seed        int64
 }
 
+var (
+	readSamplesCh          = make(chan func(), 1)
+	samplesToMinibatchesCh = make(chan func(), 1)
```

Comment:
So, the aim of `channel` here is to execute the readSamples/samplesToMinibatches lazily?

Reply:
The original `readSamples` and `samplesToMinibatches` run in goroutines, which would switch between OS threads. This behavior will cause `OpenCV` to create many threads.

This change creates a bunch of background OS threads and limits `gocv`'s calls in these threads, so we can control the  
total thread number.